### DISCUSSION TITLE
Add missing states to `alarm_control_panel` entity

### DISCUSSION THREAD

**Discussion by DeathCamel58** [home-assistant/architecture#1125] [Entity Models]
It looks like `alarm_control_panel` is missing a couple states in `VALID_STATES`. The states that seem to be missing are:

```
    STATE_ALARM_PENDING,
    STATE_ALARM_ARMING,
    STATE_ALARM_DISARMING,
```

I'd like these three states to be included to allow the `Prometheus` integration to export all possible states of an `alarm_control_panel`.

**Comment by MartinHjelmare** [comment]
Valid states is in this case used for reproducing the state. That means there needs to be a corresponding service. The mentioned states don't have a corresponding service and are not meant to have that.

Why does Prometheus need to change the reproduce state module of the alarm control panel to export the alarm control panel state?

**Reply by DeathCamel58** [reply]
Prometheus queries the attributes of a `climate`, `humidifier`, and `fan` device to get a list of modes supported, and exports a metric for each mode. However, there doesn't seem to be an equivalent for an `alarm_control_panel`, and I'm working on adding that support.

Maybe instead we could add a way to query which modes are supported for `alarm_control_panel`s in a similar way instead?

**Reply by MartinHjelmare** [reply]
Nowadays entity integrations with a defined set of string states use a StrEnum to describe this. Look at eg the lawn mover integration. The alarm control panel should probably be migrated to something similar.

This is a bit more involved change than simply adding the enum though as requiring its usage is a breaking change for existing integrations. A new property should probably be introduced at the same time, that returns the enum, and is used by the state property. Not implementing the new property could generate a warning during a deprecation period.
